propagation/b3/http.go

```diff
@@ -21,6 +21,30 @@ import (
 	"github.com/openzipkin/zipkin-go/propagation"
 )
 
+type Option func(opts *Options)
+
+type Options struct {
```

Comment:
Not sure I should call this `Options` or `InjectOptions`. WDYT @basvanbeek

Reply:
InjectOptions has my preference